trunk/src/protocol/srs_protocol_http_stack.cpp

```diff
@@ -754,6 +752,32 @@ srs_error_t SrsHttpServeMux::handle(std::string pattern, ISrsHttpHandler* handle
     return srs_success;
 }
 
+void SrsHttpServeMux::unhandle(std::string pattern, ISrsHttpHandler* handler)
+{
+    if (true) {
+        std::map<std::string, SrsHttpMuxEntry*>::iterator it = entries.find(pattern);
+        if (it != entries.end()) {
+            SrsHttpMuxEntry* entry = it->second;
+            entries.erase(it);
+
+            // We don't free the handler, because user should free it.
+            if (entry->handler != handler) {
+                srs_freep(entry);
+            }
+        }
+    }
+
+    std::string vhost = pattern;
+    if (pattern.at(0) != '/') {
```

Comment:
`vhosts` store the `vhost` -> `ISrsHttpHandler` map, but the value, `ISrsHttpHandler` never used, so is is ok to downgrade the map -> `set`?
About the code to extract the `vhost`, they are repeated in:
https://github.com/ossrs/srs/blob/e3d74fb04504cce64f78119627efe3aae825dda3/trunk/src/protocol/srs_protocol_http_stack.cpp#L706-L712

Reply:
Need another PR to refine the code, because it not about the smart ptr.